core-bundle/src/Messenger/AutoFallbackWorker.php

```diff
@@ -0,0 +1,173 @@
+            return;
+        }
+
+        foreach ($event->getWorker()->getMetadata()->getTransportNames() as $transportName) {
+            $this->ping($transportName);
+        }
+    }
+
+    #[AsEventListener]
+    public function onWorkerRunning(WorkerRunningEvent $event): void
+    {
+        // If we are running in our fallback worker process, we never ping (otherwise we
+        // would self-disable)
+        if ($this->fallbackWorkerRunning) {
+            // If we process more messages than accumulated during our request, we start the
+            // kernel.terminate worker with a --limit that is potentially higher than what's
+            // on the queue in total. This would cause the worker to run endlessly (or until
+            // reaching the time limit). This is not necessary. If we know this is our own
+            // fallback worker, and it is idle, we can stop it right away to free the process
+            // for the next requests.
+            if (self::ADDITIONAL_MESSAGES > 0 && $event->isWorkerIdle()) {
+                $event->getWorker()->stop();
+            }
+
+            return;
+        }
+
+        foreach ($event->getWorker()->getMetadata()->getTransportNames() as $transportName) {
+            $this->ping($transportName);
+        }
+    }
+
+    #[AsEventListener]
+    public function onWorkerStopped(WorkerStoppedEvent $event): void
+    {
+        $this->fallbackWorkerRunning = false;
+    }
+
+    public function ping(string $transportName): void
+    {
+        $item = $this->getCacheItemForTransportName($transportName);
+        $item->expiresAfter(new \DateInterval(self::GRACE_PERIOD));
+
+        $this->cache->save($item);
+    }
+
+    private function isWorkerRunning(string $transportName): bool
+    {
+        return $this->getCacheItemForTransportName($transportName)->isHit();
+    }
+
+    private function getCacheItemForTransportName(string $transportName): CacheItemInterface
+    {
+        return $this->cache->getItem('auto-fallback-worker-'.$transportName);
+    }
+
+    private function increaseMessageCountForTransportName(string $transportName): void
+    {
+        if (!isset($this->messagesSentDuringRequest[$transportName])) {
+            $this->messagesSentDuringRequest[$transportName] = 0;
+        }
+
+        ++$this->messagesSentDuringRequest[$transportName];
+    }
+
+    private function processTransport(string $transportName, int $limit): void
+    {
+        $this->fallbackWorkerRunning = true;
+        $input = new ArrayInput([
+            'receivers' => [$transportName],
+            '--limit' => $limit,
+            '--time-limit' => 30, // Ensure the kernel.terminate worker doesn't run forever
+            '--no-reset',
+        ]);
+
+        // No need to log anything because this is done by the messenger:consume command
+        // already and would only cause log duplication.
+        $this->consumeMessagesCommand->run($input, new NullOutput());
```

Comment:
Should/can we reset the container before executing the command here?

Reply:
Why? Symfony Messenger `reset()` on every message anyway :) See `ResetServicesListener`.

Reply:
Ahh, didn’t know that ☺️